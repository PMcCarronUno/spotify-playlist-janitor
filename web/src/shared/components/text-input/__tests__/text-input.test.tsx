import { act, fireEvent, getByTestId, render } from "@testing-library/react";

import { TextInput } from "..";

describe("<TextInput />", () => {
  let container: HTMLElement;
  let props: any;
  let mockOnChange = jest.fn();

  beforeEach(() => {
    props = {
      label: "Test Input",
      placeholder: "Enter text...",
      id: "text-input-test",
      "data-testid": "text-input-test",
      value: "",
      onChange: mockOnChange,
    };
    ({ container } = render(<TextInput {...props} />));
  });

  it("should render TextInput component", () => {
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render TextInput component with value", () => {
    ({ container } = render(
      <TextInput
        {...{
          ...props,
          value: "Text Value",
        }}
      />
    ));
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render TextInput component as password input", () => {
    ({ container } = render(
      <TextInput
        {...{
          ...props,
          type: "password",
        }}
      />
    ));
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render TextInput component as number input", () => {
    ({ container } = render(
      <TextInput
        {...{
          ...props,
          type: "number",
        }}
      />
    ));
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should execute onChange when text input", async () => {
    const input = getByTestId(container, "text-input-test");

    act(() => {
      fireEvent.change(input, { target: { value: "123" } });
    });

    expect(mockOnChange).toHaveBeenCalled();
  });
});
