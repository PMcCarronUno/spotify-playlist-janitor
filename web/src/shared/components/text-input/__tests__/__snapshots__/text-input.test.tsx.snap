// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should render TextInput component 1`] = `
<div
  class="sc-beyTiQ iQVXfo"
>
  <div
    class="input-container"
  >
    <input
      data-testid="text-input-test"
      id="text-input-test"
      placeholder="Enter text..."
      value=""
    />
    <label
      class=""
      for="text-input-test"
    >
      Test Input
    </label>
  </div>
</div>
`;

exports[`<TextInput /> should render TextInput component as number input 1`] = `
<div
  class="sc-beyTiQ iQVXfo"
>
  <div
    class="input-container"
  >
    <input
      data-testid="text-input-test"
      id="text-input-test"
      placeholder="Enter text..."
      type="number"
      value=""
    />
    <label
      class=""
      for="text-input-test"
    >
      Test Input
    </label>
  </div>
</div>
`;

exports[`<TextInput /> should render TextInput component as password input 1`] = `
<div
  class="sc-beyTiQ iQVXfo"
>
  <div
    class="input-container"
  >
    <input
      data-testid="text-input-test"
      id="text-input-test"
      placeholder="Enter text..."
      type="password"
      value=""
    />
    <label
      class=""
      for="text-input-test"
    >
      Test Input
    </label>
  </div>
</div>
`;

exports[`<TextInput /> should render TextInput component with value 1`] = `
<div
  class="sc-beyTiQ iQVXfo"
>
  <div
    class="input-container"
  >
    <input
      data-testid="text-input-test"
      id="text-input-test"
      placeholder="Enter text..."
      value="Text Value"
    />
    <label
      class="filled"
      for="text-input-test"
    >
      Test Input
    </label>
  </div>
</div>
`;
